apps/settings/js/panels/addon_details/addon_details.js

```diff
@@ -0,0 +1,169 @@
+define(function(require) {
+  'use strict';
+
+  var SettingsService = require('modules/settings_service');
+  var ManifestHelper = require('shared/manifest_helper');
+  var AppIconHelper = require('modules/app_icon_helper');
+  var AddonManager = require('modules/addon_manager');
+  var Toaster = require('shared/toaster');
+
+  function AddonDetails(panel) {
+    this.panel = panel;
+  }
```

Comment:
As the panel won't change so we could query all elements at once here instead of do the queries each time we render an app.